src/services/item/plugins/app/appData/appData.service.ts

```diff
@@ -285,4 +266,36 @@ export class AppDataService {
 
     return isValid;
   }
+
+  async upload(
+    dbConnection: DBConnection,
+    account: AuthenticatedUser,
+    file: MultipartFile,
+    item: ItemRaw,
+  ) {
+    const appDataValue = await this.appDataFileService.upload(account, file, item);
+
+    const { id: appDataId } = await this.appDataRepository.addOne(dbConnection, {
+      itemId: item.id,
+      actorId: account.id,
+      appData: appDataValue,
+    });
+
+    // get relations
+    const appData = await this.appDataRepository.getOne(dbConnection, appDataId);
```

Comment:
Why do we have to make a get afterwards, can't we return the created appData from the addOne ? This looks like legacy behaviour from TypeORM. Do we want to change it, maybe in another one.

Reply:
We need item, creator and account to be returned for compatibility, `returning` cannot give this back. Whether we really need them is another question, we can simplify afterwards.